src/System.Private.DataContractSerialization/src/Resources/System.Private.DataContractSerialization.rd.xml

```diff
@@ -10,6 +10,45 @@
           </Method>
         </Type>
         <Type Name="KeyValuePairAdapter`2" Dynamic="Required All" />
+        <!-- Reflection-based serialization requires the entries below. -->
+        <Type Name="CollectionDataContract">
+          <Type Name="CollectionDataContractCriticalHelper">
+            <Method Name="BuildCreateGenericDictionaryEnumerator{K, V}" Dynamic="Required">
+              <GenericParameter Name="K" Dynamic="Required All" />
```

Comment:
They are actually not needed. I've removed them. 

I was hoping they can include the needed metadata for IEnumerable types, but it didn't work.